File: src/main/java/ru/reosfire/wmlib/guis/Inventory/Components/RawItem.java
package ru.reosfire.wmlib.guis.Inventory.Components;

import org.bukkit.inventory.Inventory;
import ru.reosfire.wmlib.guis.Inventory.Gui;
import ru.reosfire.wmlib.text.Replacement;
import ru.reosfire.wmlib.yaml.common.gui.RawItemConfig;

public class RawItem extends GuiComponent
{
    private final RawItemConfig Config;
    public RawItem(RawItemConfig config, Gui gui)
    {
        super(gui);
        Config = config;
    }
    public void RenderTo(Inventory inventory, Replacement... replacements)
    {
        for (Integer index : Config.Indexes)
        {
            inventory.setItem(index, Config.Item.Unwrap(gui.Player, replacements));
        }
    }

    @Override
    public void Register()
    {

    }
}

File: src/main/java/ru/reosfire/wmlib/guis/inventory/components/RawItem.kt
package ru.reosfire.wmlib.guis.inventory.components

import org.bukkit.inventory.Inventory
import ru.reosfire.wmlib.guis.inventory.Gui
import ru.reosfire.wmlib.text.Replacement
import ru.reosfire.wmlib.yaml.common.gui.RawItemConfig

class RawItem(private val Config: RawItemConfig, gui: Gui) : GuiComponent(gui) {
    override fun renderTo(inventory: Inventory, vararg replacements: Replacement) {
        for (index in Config.indexes!!) {
            inventory.setItem(index, Config.item.unwrap(gui.Player, *replacements))
        }
    }

    override fun register() {}
}